src/o365/graph/commands/report/report-teamsdeviceusageusercounts.ts

```diff
@@ -0,0 +1,108 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import GraphCommand from "../../GraphCommand";
+import request from '../../../../request';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  period: string;
+}
+
+class GraphReportTeamsDeviceUsageUserCountsCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.REPORT_TEAMSDEVICEUSAGEUSERCOUNTS}`;
+  }
+
+  public get description(): string {
+    return 'Get the number of Microsoft Teams daily unique users by device type';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.period = typeof args.options.period !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): Promise<{}> => {
+        const endpoint: string = `${auth.service.resource}/v1.0/reports/getTeamsDeviceUsageUserCounts(period='${encodeURIComponent(args.options.period)}')`;
+
+        const requestOptions: any = {
+          url: endpoint,
+          headers: {
+            authorization: `Bearer ${auth.service.accessToken}`
+          }
+        };
+
+        return request.get(requestOptions);
+      })
+      .then((res: any): void => {
+        cmd.log(res);
+
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-p, --period <period>',
+        description: 'The length of time over which the report is aggregated. Supported values D7|D30|D90|D180',
+        autocomplete: ['D7', 'D30', 'D90', 'D180']
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.period) {
+        return 'You can\'t run this command without period parameter.';
+      }
+
+      if (args.options.period) {
```

Comment:
This check is not needed since the previous condition fulfills the check for `period` being mandatory.